### ISSUE TITLE
[Bug] TotK upscaling issues

### ISSUE THREAD

**Issue by alexkiri** [Ryujinx/Ryujinx#5269]
### Description of the issue

The self shadowing effect looks broken / distracting at Resolution Scale 2x and beyond, especially in motion, as you can see in the clip below:
Before: https://streamable.com/vsqa01
I have tracked the texture used in this effect, and it seems it's not upscaled because, as mentioned in TextureCache.cs:186 "Targets that are roughly 16:9 can only be rescaled if they're equal to or above 360p"
By adding a dirty fix to skip this condition, I managed to improve this effect, but it completely breaks other effects, such as Ambient Occlusion.
After: https://streamable.com/f2jkr2
Is there any way to improve the upscaling logic?

The videos above, in full res: https://we.tl/t-6AGc7jbAhW

### Reproduction steps

The effect it's mostly noticeable on Link and other NPCs when:
- there is a strong light source nearby, such as a lamp, a lava pool
- it's dark
- indoors
- by slowly moving the camera

### Log file

[Ryujinx_1.1.872_2023-06-09_12-12-33.log](https://github.com/Ryujinx/Ryujinx/files/11704469/Ryujinx_1.1.872_2023-06-09_12-12-33.log)

### OS

Windows 10.0.22621 (X64)

### Ryujinx version

1.1.872

### Game version

1.1.2

### CPU

AMD Ryzen 7 3700X

### GPU

Nvidia RTX 3060 Ti 8GB

### RAM

16GB

### List of applied mods

1080p All In One Ryujinx v1.1.0-1.1.2
60fps_stable_static_totk
DynamicFPSv1.5.2
UI Blackscreen Fix

**Comment by MikePrime13** [comment]
I also have observed this issue in the Gerudo caves area where there are a lot of "natural" lighting in effect, i.e. torches mounted on pillars and/or walls.

Running the game at native resolution appears to fix the issue. Also, in my case, the problem is a bit intermittent and depends on the upscaling method. Bilinear appears to be affected the most, while FSR is not affected, but not desirable because the upscaling creates a new look and feel that is different (and subjectively undesirable) from bilinear upscaling which is closer to the native presentation.

Hardware Profile 1:
Intel 8700k/RTX4070 @ Ryujinx 1.1.872

Hardware Profile 2:
AMD Ryzen 3600K/RTX2070 Super @ Ryujinx 1.1.872

Applied Mods:
30fps_stable_static_totk (no effect whether on/off)
DynamicFPSv1.5.2 (no effect whether on/off)
UI Blackscreen fix (no effect whether on/off -- tested for both enabled/disabled)
Chuck Visual Fix 1.12 (Disable FSR downscaling; Disable Quality Reduction when framerate dips; Shadows 1024x1024; Force 1080p) -- enabled/disabled but no difference meaning that error is still present.

The problem is very pronounced when moving from room to room and/or moving around a pillar where there is a lightsource from the other side -- the shadows render and shown while the camera is moving but disappears and/or glitches when the camera stops moving.

**Comment by alexkiri** [comment]
As described initially, it only appears at Resolution Scale 2x and up, at native resolution it won't appear at all. It also strongly depends on the light source. Bilinear / FSR has nothing to do with this issue.

**Comment by MikePrime13** [comment]
I respectfully disagree regarding your conclusion that Bilinear/FSR have nothing to do with the issue without any further explanation -- I am merely reporting what I am observing on my end, and in my case, the issue sometimes goes away when I switch the upscaling technique from bilinear to FSR. I don't know whether this is something that occurs on your end as well, but I am simply reporting/providing data as to what I have observed in not one, but two PCs that I regularly use for playing TOTK.

Again, I am doing this in good faith to report what I observe to be happening, and if this proves later to be irrelevant, then I am happy to be wrong. If you could kindly explain to me as to how you came to the conclusion that this issue has nothing to do with Bilinear/FSR settings, I am happy to agree with your conclusion. 

From my limited and layperson understanding, the upscaling techniques apply directly to the textures, and therefore perhaps there is potentially a function and/or setting that gets affected when you switch the upscaling techniques in real time that seems to trigger/resolve the issue, and I have observed this in two separate NVIDA machines. That is my reasoning to reply to your original post to add another potential data point for this issue.

Cheers.

**Comment by alexkiri** [comment]
Thanks for the detailed answer. I must insist on my previous conclusion. There is some confusion regarding these 2 different settings: 
- Resolution Scale is applied to render targets, that means all the intermediate renders that render each effect (such as shadows, self shadowing, ambient occlusion, volumetric light, etc)
- Upscale is applied to the framebuffer, that is the final render that is displayed in the emulator window. This upscale is performed once on the framebuffer, at the end of the rendering. The difference between Bilinear and FSR can best be observed on a 4k monitor, with Resolution Scale set to native.

None of these settings apply to the raw textures. What the game contains is what is used in the rendering, no upscaling of individual textures is performed.

One additional argument for the irrelevance of the Upscale Bilinear/FSR setting. If a game renders native 1080, and you set Resolution Scale 2x, and you display it in fullscreen on a native 4k display, it will look identical, regardless of the settings.

Sorry for the confusion, I hope this is clearer.

**Comment by MikePrime13** [comment]
Thank you so much for the clarification. I am so glad you handle my response well.

Based on your response, I fully defer to you and leave it up to you whether we leave my previous comment intact, or remove it as to not confuse people reviewing the issue ticket. I don't want to create any further debate on this because I have no technical expertise on this matter, and I understand and agree with your clear and concise explanations why it's irrelevant.

Much appreciated.
